feature_engine/datetime/datetime.py

```diff
@@ -250,13 +275,36 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         # Check if input data contains same number of columns as dataframe used to fit.
         _check_input_matches_training_df(X, self.n_features_in_)
 
+        # special case index
+        if self.variables_ == ["index"]:
+            if self.missing_values == "raise":
+                if X.index.isnull().any():
+                    raise ValueError(
+                        "Index contain NaN. Check and "
+                        "remove those before using this transformer."
```

Comment:
could we rephrase as in fit()? 

Also wondering, we are repeating code in fit and transform, should we capture it in a private method in this class? 

```
       if self.missing_values == "raise":
            self._check_contains_na()
```